scripts/apidocs-gen/post-process-types.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This script post-processes generated API reference docs to replace links to
+# external SDK types with their primitive equivalents, so the docs do not
+# reference missing type definitions.
+#
+# It operates in-place on the generated markdown files under ./docs/.
+
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/../.."
+
+# Candidate docs to process (some may not exist in all builds)
+DOC_FILES=(
+  "${SCRIPT_ROOT}/docs/konnect-api-reference.md"
+  "${SCRIPT_ROOT}/docs/all-api-reference.md"
+  "${SCRIPT_ROOT}/docs/configuration-api-reference.md"
+  "${SCRIPT_ROOT}/docs/gateway-operator-api-reference.md"
+  "${SCRIPT_ROOT}/docs/incubator-api-reference.md"
+)
+
+for f in "${DOC_FILES[@]}"; do
+  if [[ -f "${f}" ]]; then
```

Comment:
IMHO we shouldn't check it here but rather return with an error. The `DOC_FILES` array should contain only the files that actually **do exist**, right?